integrations/spark/integrations/sparksql/1.json

```diff
@@ -1,65 +1,176 @@
+                  "nullable": true,
+                  "metadata": {}
+                }
+              ]
+            },
+            "deserializer": [
+              {
+                "class": "org.apache.spark.sql.catalyst.expressions.objects.Invoke",
+                "num-children": 1,
+                "targetObject": 0,
+                "functionName": "toString",
+                "dataType": "object",
+                "arguments": [],
+                "propagateNull": true,
+                "returnNullable": false
+              },
+              {
+                "class": "org.apache.spark.sql.catalyst.expressions.Cast",
+                "num-children": 1,
+                "child": 0,
+                "dataType": "string"
               },
-              "type" : "org.apache.spark.sql.execution.datasources.LogicalRelation",
-              "name" : "LogicalRelation"
-            } ]
-          } ]
-        } ]
+              {
+                "class": "org.apache.spark.sql.catalyst.expressions.AttributeReference",
+                "num-children": 0,
+                "name": "value",
+                "dataType": "string",
+                "nullable": true,
+                "metadata": {},
+                "qualifier": []
+              }
+            ],
+            "child": 0
+          },
+          {
+            "class": "org.apache.spark.sql.catalyst.plans.logical.Project",
+            "num-children": 1,
+            "projectList": [
+              [
+                {
+                  "class": "org.apache.spark.sql.catalyst.expressions.AttributeReference",
+                  "num-children": 0,
+                  "name": "value",
+                  "dataType": "string",
+                  "nullable": true,
+                  "metadata": {},
+                  "qualifier": []
+                }
+              ]
+            ],
+            "child": 0
+          },
+          {
+            "class": "org.apache.spark.sql.execution.datasources.LogicalRelation",
+            "num-children": 0,
+            "relation": null,
+            "output": [
+              [
+                {
+                  "class": "org.apache.spark.sql.catalyst.expressions.AttributeReference",
+                  "num-children": 0,
+                  "name": "value",
+                  "dataType": "string",
+                  "nullable": true,
+                  "metadata": {},
+                  "qualifier": []
+                }
+              ]
+            ],
+            "isStreaming": false
+          }
+        ]
       }
     }
   },
-  "job" : {
-    "namespace" : "ns_name",
-    "name" : "job_name"
+  "job": {
+    "namespace": "ns_name",
+    "name": "job_name"
   },
-  "inputs" : [ {
-    "namespace" : "gs.bucket",
-    "name" : "gs://bucket/data.txt"
-  } ],
-  "outputs" : [ ],
-  "producer" : "https://github.com/OpenLineage/OpenLineage/blob/v1-0-0/client"
+  "inputs": [
+    {
+      "namespace": "file://",
```

Comment:
Can we drop the trailing `://` part? It seems implied.

Reply:
This is derived from the dataset URI. Running in a spark cluster in the cloud, it would be read from a URI like `gs://bucket/path/to/file`. Running locally, it's `file:///path/to/file`. In those other cases, the namespace really is `gs://bucket`- it's that in the `file://` URI, the authority is blank.